server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeSort.java

```diff
@@ -158,6 +193,29 @@ public Double apply(@Nonnull ComponentDtoWithSnapshotId input) {
     }
   }
 
+  private static class ComponentDtoWithSnapshotIdToMeasureVariationValue implements Function<ComponentDtoWithSnapshotId, Double> {
+    private final MetricDto metric;
+    private final Table<String, MetricDto, MeasureDto> measuresByComponentUuidAndMetric;
+    private final int variationIndex;
+
+    private ComponentDtoWithSnapshotIdToMeasureVariationValue(@Nullable MetricDto metric,
+      Table<String, MetricDto, MeasureDto> measuresByComponentUuidAndMetric, int variationIndex) {
+      this.metric = metric;
+      this.measuresByComponentUuidAndMetric = measuresByComponentUuidAndMetric;
+      this.variationIndex = variationIndex;
+    }
+
+    @Override
+    public Double apply(@Nonnull ComponentDtoWithSnapshotId input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)